mmcv/cnn/bricks/scale.py

```diff
@@ -19,3 +19,35 @@ def __init__(self, scale: float = 1.0):
 
     def forward(self, x: torch.Tensor) -> torch.Tensor:
         return x * self.scale
+
+
+class LayerScale(nn.Module):
+    """LayerScale layer.
+
+    Args:
+        dim (int): Dimension of input features.
+        inplace (bool): Whether performs operation in-place.
+            Default: `False`.
+        data_format (str): The input data format, could be 'channels_last'
+            or 'channels_first', representing (B, C, H, W) and
+            (B, N, C) format data respectively. Default: 'channels_last'.
+    """
+
+    def __init__(self,
+                 dim: int,
+                 inplace: bool = False,
+                 data_format: str = 'channels_last'):
+        super().__init__()
+        assert data_format in ('channels_last', 'channels_first'), \
+            "'data_format' could only be channels_last or channels_first."
+        self.inplace = inplace
+        if data_format == 'channels_first':
+            self.weight = nn.Parameter(torch.ones(dim, 1, 1) * 1e-5)
```

Comment:
shape (dim, 1, 1) seems inflexible, could we view the tensor as different shape according to `data_format` in `forward` ?

Reply:
You mean following is better? I fixed based on zhouzaida's review.

```
def forward(self, x):
        if self.data_format == 'channels_first':
            if self.inplace:
                return x.mul_(self.weight.view(-1, 1, 1))
            else:
                return x * self.weight.view(-1, 1, 1)
        return x.mul_(self.weight) if self.inplace else x * self.weight
```

Reply:
Ahhh, If we use `self.weight.view(-1, 1, 1)`, it is proper to implement this logic in `__init__`. I mean that if you want the view of `self.weight` could adjust with the shape input input `x`, you can write it as follows:

```python
        if self.data_format == 'channels_first':
            shape =  tuple((-1, *(1 for _ in range(x.dim()-1))))
        else:
            shape =  tuple((*(1 for _ in range(x.dim()-1)), -1))
        if self.data_format == 'channels_first':
            if self.inplace:
                return x.mul_(self.weight.view(*shape))
            else:
                return x * self.weight.view(*shape)
```

And then, `x` could be a tensor with any dim.

Reply:
I understand and have fixed it.